libbeat/beat/pipeline.go

```diff
@@ -60,6 +61,10 @@ type ClientConfig struct {
 
 	// ClientListener configures callbacks for monitoring pipeline clients
 	ClientListener ClientListener
+
+	// InputMetricsRegistry is the metrics registry used to aggregate pipeline
+	// metrics per input.
+	InputMetricsRegistry *monitoring.Registry
```

Comment:
This field confuses me, I thought part of what we talked about for this refactor was that the input itself shouldn't have to track its own registry since the pipeline can handle it. This is confusing since it doesn't explain what the contract is for this field or when / why it should be provided, but beyond that it seems like we don't really want people to use it as "Config" since they should never create their own custom values, right? I think passing through the registry from the input should be done in the pipeline instead of adding new requirements to the input API contract.

Reply:
This is what will pass the metrics registry from the input to the pipeline client.

> I think passing through the registry from the input should be done in the pipeline instead of adding new requirements to the input API contract.

Where in the pipeline then? Now I'm the one confused...

The pipeline itself, AFAICS is for the whole beat, or beat.Beat instance, therefore not input specific or aware. The pipeline client on the other hand is built for a specific input. That's why it's there, to pass the metrics registry from the input context to the pipeline client. And then later to the output.

Reply:
"Handled by the pipeline" in this case I mean creating the registry should be the responsibility of whoever calls `input.Run`, potentially propagated by an internal field in `Context` (that the input itself doesn't have to know about). Part of the v2 input framework, not something the input needs to pass through manually.

Reply:
There should be a way to avoid it, I will give an example draft of how to do it shortly